Refactored SpotifyUri Parsing

philippdrebes opened the pull request:
This is a refactored version of the SpotifyUri class. The old implementation did work with most URI's but not with all variations (e.g. playlists).

Tested with the following URI configurations:
spotify:track:[id]
spotify:artist:[id]
spotify:album:[id]
spotify:user:[id]
spotify:user:[id]:playlist:[id]

philippdrebes commented:
Seems like the ci build cannot handle C# 7 features, such as the inline declaration of variables inside TryParse.
Therefore the failed ci check was caused by these lines of code in SpotifyUri.cs:
[line 43](https://github.com/philippdrebes/SpotifyAPI-NET/blob/cdcfdf066a001aea3073be5b6fd68c2b90fe8c18/SpotifyAPI/Local/Models/SpotifyUri.cs#L43) 
[line 50](https://github.com/philippdrebes/SpotifyAPI-NET/blob/cdcfdf066a001aea3073be5b6fd68c2b90fe8c18/SpotifyAPI/Local/Models/SpotifyUri.cs#L50)

for further reference: [What’s New in C# 7.0](https://blogs.msdn.microsoft.com/dotnet/2016/08/24/whats-new-in-csharp-7-0/)

JohnnyCrazy commented:
Thanks @philippdrebes , looks very nice! 👍 
 
Would you mind replacing the 'new' inline `TryParse`s with the old method? This way we don't have to update the whole project to vs2017. I want to do this at some point, but not right now. 

Also, you updated JSON.NET in the main project, but not in the test project, which leads to some warnings. Feel free to updated all the deps in the test project as well (But I can also do that, just tell me what you prefer 👍 )

Thanks again!

philippdrebes commented:
Hi @JohnnyCrazy 
The inline TryParse's are now replaced with the "old" method (pre C# 7).
Also, the NuGet packages inside of the test project are now updated.

JohnnyCrazy commented:
Perfect, a new release will follow soon, Thanks! :wink:
